test/odbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,361 @@
+    insert_string += comma + "(" + std::to_string(i) + "," + valid_inserted_values[i] + ")";
+    comma = ",";
+  }
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+ 
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(affected_rows, valid_inserted_values.size());
+  
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < valid_inserted_values.size(); ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+
+    if (valid_inserted_values[i] != "NULL")
+    {
+      ASSERT_EQ(data_len, BYTES_EXPECTED);
+      ASSERT_EQ(data, StringToBigInt(valid_inserted_values[i]));
+    }
+    else 
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+  odbcHandler.CloseStmt();
+
+}
+
+TEST_F(PSQL_DataTypes_Bigint, Insertion_Fail) {
+
+  const int BYTES_EXPECTED = 8;
+
+  long long int pk;
+  long long int data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector <string> invalid_inserted_values = {
+    "-9223372036854775809",
+    "9223372036854775808",
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns = {
+    {1, SQL_C_SBIGINT, &pk, 0, &pk_len},
+    {2, SQL_C_SBIGINT, &data, 0,  &data_len}
+  };
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Attempt to insert values that are out of range and assert that they all fail
+  for (int i = 0; i < invalid_inserted_values.size(); i++) {
+
+    string insert_string = "(" + std::to_string(i) + "," + invalid_inserted_values[i] + ")";
+
+    rcode = SQLExecDirect(odbcHandler.GetStatementHandle(), (SQLCHAR*) InsertStatement(TABLE_NAME, insert_string).c_str(), SQL_NTS);
+    ASSERT_EQ(rcode, SQL_ERROR);
+  }
+
+  // Select all from the tables and assert that nothing was inserted
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+}
+
+TEST_F(PSQL_DataTypes_Bigint, Update_Success) {
+
+  const string PK_INSERTED = "1";
+  const string DATA_INSERTED = "1";
+  const string DATA_UPDATED_VALUE = "5";
```

Comment:
the 5 is fine, but could potentially use one of the min or max values  as in 
 "-9223372036854775808",
    "9223372036854775807"
to update or both (do two update statements) or three (5 and the min and max)

Reply:
I can expand this to 3 update statements for more test coverage.